### PR TITLE
Improve API publishing

### PR DESCRIPTION
**Pull request by tharindu1st**
Reducing no of Service calls to create and publish API

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/service/APILocalEntryAdmin.java [hunk 1]
```diff
@@ -63,7 +63,7 @@ protected LocalEntryClient getLocalEntryAdminClient(String tenantDomain) throws
      * @throws AxisFault
      */
     public Object getEntry(String key, String tenantDomain) throws AxisFault {
-        LocalEntryClient localEntryAdminClient = getLocalEntryAdminClient(tenantDomain);
+        LocalEntryServiceProxy localEntryAdminClient = getLocalEntryAdminClient(tenantDomain);
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/service/APILocalEntryAdmin.java]: let's rename the variable names too to replace client with proxy

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/EndpointAdminServiceProxy.java [hunk 2]
```diff
@@ -49,30 +51,20 @@ public EndpointAdminServiceClient() throws AxisFault {
...
      */
     public boolean addEndpoint(String endpointData) throws AxisFault {
+
         try {
-            return endpointAdminStub.addEndpoint(endpointData);
+            if (MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                return endpointAdmin.addEndpoint(endpointData);
+            } else {
+                return endpointAdmin.addEndpointForTenant(endpointData, tenantDomain);
+
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/EndpointAdminServiceProxy.java]: extra new line

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/EndpointAdminServiceProxy.java [hunk 3]
```diff
@@ -85,57 +77,37 @@ public boolean deleteEndpoint(String endpointName) throws AxisFault {
         try {
             // This check was added due to endpoint was saved in endpoint directory instead of saving inline with api
             // synapse file.
-            if (checkEndpointExistBeforeDelete(endpointName, null)) {
-                return endpointAdminStub.deleteEndpoint(endpointName);
+            if (MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)){
+                return endpointAdmin.deleteEndpoint(endpointName);
+            }else{
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/EndpointAdminServiceProxy.java]: needs formatting

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/MediationSecurityAdminServiceProxy.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2005-2013, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/MediationSecurityAdminServiceProxy.java]: incorrect copyright

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/dto/CredentialDto.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+
+package org.wso2.carbon.apimgt.api.dto;
```
Comment (ruks) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/dto/CredentialDto.java]: class-level comments are missing, and there are additional white spaces that need to be removed.

**Pull request merged**